### PR TITLE
Additions from ROpenSci Unconference (Spring 2016)

### PR DESCRIPTION
**Pull request by geanders**
This pull requests incorporates work started at the spring's ROpenSci Unconference. Specific elements of this pull request are:

* Vignette showing how to use `rnoaa` to pull weather data that can then be used in conjunction with air pollution data pulled using OpenAQ ("rnoaa_openaq.RMD").
* Added text to help files for ghcnd_* family of functions to provide users more details and clarify differences between different functions in the family (all associated change are in the "ghcnd.R" file).
* Added `meteo_nearby_stations` function, to identify stations near a latitude-longitude pair for Global Historical Climatology Network daily weather data (defined in "meteo_distance.R" file).
* Added `meteo_pull_monitors`, to pull Global Historical Climatology Network daily weather data from multiple monitors with a single function call (defined in "helpers_ghcnd.R" file).
* Added `meteo_show_cache` and `meteo_clear_cache` functions, for cache management (both files in "meteo_cache.r") (other functions we're adding do not rely on these functions).
* Added `meteo_coverage` function to determine the coverage (percent of non-missing weather observations over dates pulled) for each weather monitor (in "meteo_utils.R" file), with its own method for plotting coverage. 
* Added `vis_miss` function to visualize missingness in a dataframe, including weather dataframes generated with `rnoaa`
* Various helper functions to use with the above new functions.
* Added basic tests for the two main functions added in this PR (in "test-meteo.R" file)

### CODE REVIEW

File: vignettes/rnoaa_openaq.Rmd [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  ungroup() %>%
+  rename(id = location) %>%
+  mutate(id = factor(id))
+
+stationsDelhi <- meteo_nearby_stations(lat_lon_df = as.data.frame(lat_lon_df),
+                                       station_data = station_data,
+                                       radius = 15,
+                                       year_min = 2016,
+                                       var = c("TAVG", "PRCP"))
+stationsDelhi <- unique(rbind_all(stationsDelhi) %>% select(- distance))
```
Comment (sckott) [inline vignettes/rnoaa_openaq.Rmd]: @geanders switch to use `bind_rows` instead, as `rbind_all` is deprecated, and change in import statement in `rnoaa-package`

File: R/meteo_utils.r [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      col_cov <- setNames(cbind.data.frame(col_cov, stringsAsFactors=FALSE), meteo_cols)
+      dplyr::bind_cols(dat, col_cov)
+    }) -> out
+  class(out) <- c("meteo_coverage", class(out))
+  if (verbose) return(invisible(out))
+  out
+}
+
+#' @export
+autoplot.meteo_coverage <- function(df) {
```
Comment (sckott) [inline R/meteo_utils.r]: @geanders  check says 

```r
W  checking for missing documentation entries
   Undocumented code objects:
     ‘autoplot.meteo_coverage’
```

perhaps need `@rdname meteo_coverage` below the `@export` line

**Comment by sckott** [comment]
@geanders  thanks for the PR - made a few comments for changes

**Comment by geanders** [comment]
Thanks so much for taking the time to review. I've made the suggested changes.

**Comment by sckott** [comment]
thanks @geanders merged

**Comment by geanders** [comment]
Thank you so much! I really appreciate you taking the time on this pull request.


Would you let me know the next time you submit a version of rnoaa to CRAN? I have a student working on a package that builds on some of these functions, so it will be good to know for getting the dependencies straight in her package code.


Best,

Brooke

________________________________

**Comment by sckott** [comment]
thanks for all your work on this, and the work of others, i'll add you as a contributor

**Pull request merged**